internal/runbits/buildscript/merge.go

```diff
@@ -0,0 +1,163 @@
+package buildscript
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/multilog"
+	"github.com/ActiveState/cli/pkg/localcommit"
+	bpModel "github.com/ActiveState/cli/pkg/platform/api/buildplanner/model"
+	"github.com/ActiveState/cli/pkg/platform/api/mono/mono_models"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildexpression"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildscript"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+func Merge(proj *project.Project, remoteCommit strfmt.UUID, strategies *mono_models.MergeStrategies, auth *authentication.Auth) error {
```

Comment:
This feels like it's doing too much for a runbit. Can we move the merge specific logic into its own package (or maybe make it part of the buildexpression?). The runbit shouldn't contain business logic, just direct input and business logic.

I'll defer the rest of the review for the next iteration as I imagine this change would be significant.

Reply:
Sure. I moved it into a subpackage of buildexpression to avoid an import cycle.